src/components/TransactionResult.tsx

```diff
@@ -0,0 +1,53 @@
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  AlertTitle,
+  Flex,
+} from "@chakra-ui/react";
+
+export enum ActionName {
+  Deposit = "Deposit",
+  Withdraw = "Withdraw",
+  Approve = "Approve",
+  Associate = "Associate",
+  Mint = "Mint",
+  Claim = "Claim",
+}
+
+const getAlertTitle = (isSuccess = true, actionType?: ActionName) => {
+  return `${actionType} ${isSuccess ? "success" : "error"}`;
+};
+
+export const TransactionResult = ({
+  actionName,
+  transactionResult,
+  transactionError,
+}: {
+  actionName: ActionName | undefined;
+  transactionResult: `0x${string}` | undefined;
+  transactionError: Error | null | undefined;
```

Comment:
docs say its either error or null, but not undefined?

error: null | TError
Defaults to null
The error object for the query, if an error was thrown.